### PR TITLE
Sonar cloud and clean up

### PR DESCRIPTION
**Pull request by apple00juice**
I continued working on feature-gameView by mistake. Thus merge into feature-gameView, else all the tiles are shown again ^^ (If this gets approved I will merge this after feature-gameview)
I wanted to create a popUp with a backdrop and make it more general, thus I created a new one based on luis's initial design. Just some example popUps:
NameErrorupdatingScreenshot 2022-04-10 160052

I also looked at sonarCloud and fixed most of the bugs/code smells and warning/errors in the console and let the error messages be displayed correctly after an API call. I changed stuff in HomePage, PublicLobbies, CreateLobby, Lobby, UpdateLobby if I remember correctly.
The design stayed the same, I just reduced the margin of the title in the create/update view because it pushed the buttons off the screen.

### CODE REVIEW

File: src/components/views/CreateLobby.js [hunk 1]
```diff
@@ -54,31 +49,21 @@ const CreateLobby = props => {
...
+                        headers: {'token': ''}
                     }
                 );
 
-                var resStatus = response.status;
                 const lobbyId = response.data.lobby.id;
 
                 history.push({
                     pathname: '/lobby/' + lobbyId,
                     state: response.data
```
Comment (paolovic) [inline src/components/views/CreateLobby.js:60]: we shouldn't pass all the lobby information from frontend view to frontend view
(The same goes for the transition Lobby->UpdateLobby). We should get the information from the server instead. 
I implemented it in my branch feature_invcode_views_paul for the Lobby view. This should be merged somehow imo. (The UpdateLobby has to be adapted analogously, the UseEffect() method in UpdateLobby is missing completely so far...)

File: src/components/routing/routers/AppRouter.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import CreateLobby from "components/views/CreateLobby";
 import HomePage from "components/views/HomePage";
 import PublicLobbies from "components/views/PublicLobbies";
 import UpdateLobby from "components/views/UpdateLobby";
-import Header from "components/views/Header"
+import Header from "components/ui/Header"
```
Comment (paolovic) [inline src/components/routing/routers/AppRouter.js:8]: I implemented the skeleton for nested routing in my branch feature_invcode_views_paul. This should be merged somehow imo.

**Comment by apple00juice** [comment]
Will wait until luis's merged his branch

File: src/components/views/Lobby.js [hunk 3]
```diff
@@ -25,102 +25,135 @@ const Lobby = props => {
...
+    else
+        return (
+            <BaseContainer>
+                <div className="lobby">
+                    <label className="lobby lobby-title">Lobby Information</label>
+                    <Link
+                        className="lobby link"
+                        to={{
+                            pathname: '/update-lobby/' + lobbyData.id,
+                            state: {
```
Comment (paolovic) [inline src/components/views/Lobby.js:83]: see above.

File: src/components/views/Lobby.js [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
 
-const Lobby = props => {
+const Lobby = () => {
 
     const history = useHistory();
 
     //we get the information from the creation page
     const location = useLocation();
-    const [token, setToken] = useState(location.state.token);
+    const [token] = useState(location.state.token);
```
Comment (paolovic) [inline src/components/views/Lobby.js:16]: token shouldn't be a state, should be loaded from the localStorage.getItem("token")

File: src/components/views/PublicLobbies.js [hunk 5]
```diff
@@ -1,94 +1,139 @@
...
             } catch (error) {
-                console.error(`Something went wrong while fetching the users: \n${handleError(error)}`);
-                console.error("Details:", error);
-                //alert("Something went wrong while fetching the users! See the console for details.");
+                setErrorMessage("Something went wrong while fetching the lobbies")
             }
         }
 
-        fetchData();
+        fetchData().then();
```
Comment (paolovic) [inline src/components/views/PublicLobbies.js:47]: you want to execute UseEffect synchronously and calling therefore fetchData().then()?
I am not sure, but I think you could remove the .then() here and the "async" in line 40 and you'll have the same effect.

**Review by paolovic — COMMENTED** [review]
@luis-tm will merge the parts of my branch he can use into his branch and then his into the master.
afterwards this branch can be merged into master.

**Pull request closed without merging**